ui_framework/components/card/_card_group.scss

```diff
@@ -1,23 +1,18 @@
 /**
- * 1. Use this container to offset the spacing between wrapped cards.
- */
-.kuiCardGroupContainer {
-  margin: -$cardSpacing; /* 1 */
-}
-
-/**
  * 1. Wrap cards when necessary.
+ * 2. Offset the spacing between wrapped cards.
  */
 .kuiCardGroup {
   display: flex;
   flex-wrap: wrap; /* 1 */
+  margin: -$cardSpacing; /* 2 */
 
   /**
    * 1. Use the defined width of the card to determine when to wrap.
    * 2. Use an even margin all around the card so that the spacing is still even when wrapped.
    */
   .kuiCard {
```

Comment:
You're absolutely right... this is a dilemma I've encountered several times. For context, this component was originally implemented the way you suggest, and I only recently changed it to use the current pattern (https://github.com/elastic/kibana/pull/12183/files#diff-5b434436eea8c6ef3f9c1f64780bb791L3 for reference).

There are two problems I've encountered when strictly applying BEM in this kind of situation:

1. Mixing classes is sometimes required on many elements of a single component, e.g. in the link I shared you not only need `.kuiCard` mixed with `.kuiCardGroup__card` but you also need `.kuiCard__description` mixed with `.kuiCardGroup__cardDescription`. From the point of view of someone applying the classes, this "interface" can be complex and confusing.
2. Converting these components to React then requires passing in these classes manually, as you can see in [the first commit of this PR](https://github.com/elastic/kibana/pull/12197/commits/db7120291cd949d073d7851f472316e9917d80d3#diff-e788acde9ae4ac301ec6ecb8aa048cbfR86). An alternative would be to create a wrapper component which applies this class for you. But again, I feel like both cases are added complexity in the interface.

To me, it comes down to a trade-off: sacrificing BEM in exchange for a more ergonomic interface. Personally, I think we should try to stick to BEM as much as possible, until we discover some pain in the interface. What are your thoughts on this?

Also, do you think we should document your links and these lines of reasoning in a style guide?